src/components/SearchForm/SearchForm.js

```diff
@@ -1,25 +1,21 @@
 import { useState, useContext } from "react";
 import NewsSearchContext from "../../contexts/NewsSearchContext.js";
 import HasSearchedContext from "../../contexts/HasSearchedContext.js";
+import KeywordsContext from "../../contexts/KeywordsContext.js";
+import { useForm } from "../hooks/useForm.js";
 
 function SearchForm() {
   const { handleNewsSearch } = useContext(NewsSearchContext);
   const { setHasSearched } = useContext(HasSearchedContext);
-  const [searchInput, setSearchInput] = useState("");
-  const [keywords, setKeywords] = useState([]);
-
-  console.log(keywords);
+  const { setKeyword } = useContext(KeywordsContext);
+  const { values, setValues, handleChange } = useForm({ search: "" });
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    handleNewsSearch(searchInput);
+    handleNewsSearch(values.search);
     setHasSearched(true);
-    setKeywords(searchInput);
-    setSearchInput("");
-  };
-
-  const handleInputChange = (e) => {
-    setSearchInput(e.target.value);
+    setKeyword(values.search);
+    setValues({ search: "" });
```

Comment:
`    setValues({ search: "" });
`

You need to **reset** the form only when you **open** it not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more